brut.j.dir/src/main/java/brut/directory/ZipUtils.java

```diff
@@ -0,0 +1,83 @@
+/**
+ *  Copyright (C) 2017 Ryszard Wiśniewski <brut.alll@gmail.com>
+ *  Copyright (C) 2017 Connor Tumbleson <connor.tumbleson@gmail.com>
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package brut.directory;
+
+import brut.common.BrutException;
+import brut.util.BrutIO;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+
+import java.io.*;
+import java.util.Collection;
+import java.util.zip.CRC32;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public class ZipUtils {
+
+    private static Collection<String> mDoNotCompress;
+
+    public static void zipFolders(final File folder, final File zip, final File assets, final Collection<String> doNotCompress)
+            throws BrutException, IOException {
+
+        mDoNotCompress = doNotCompress;
+        ZipOutputStream zipOutputStream = new ZipOutputStream(new FileOutputStream(zip));
+        zipFolders(folder, zipOutputStream);
+
+        // We manually set the assets because we need to retain the folder structure
+        if (assets != null) {
+            processFolder(assets, zipOutputStream, assets.getPath().length() - 6);
+        }
+        zipOutputStream.close();
+    }
+
+    private static void zipFolders(final File folder, final ZipOutputStream outputStream)
+            throws BrutException, IOException {
+        processFolder(folder, outputStream, folder.getPath().length() + 1);
+    }
+
+    private static void processFolder(final File folder, final ZipOutputStream zipOutputStream, final int prefixLength)
```

Comment:
This method is a near clone of a previous in our AndrolibResources, probably can "helperize" this

Reply:
gonna hold on this change, a bit different in how these methods operate